lib/modules/platform/bitbucket/index.ts

```diff
@@ -179,7 +183,21 @@ export async function initRepo({
         )
       ).body
     );
-    config.defaultBranch = info.mainbranch;
+
+    if (bbUseDevelopmentBranch) {
+      // Fetch Bitbucket development branch
+      developmentBranch = (
+        await bitbucketHttp.getJson<RepoBranchingModel>(
+          `/2.0/repositories/${repository}/branching-model`
+        )
+      ).body.development?.branch?.name;
+
+      mainBranch = developmentBranch ?? info.mainBranch;
+    } else {
+      mainBranch = info.mainBranch;
+    }
+
+    config.defaultBranch = mainBranch;
```

Comment:
```suggestion
    config.defaultBranch = mainBranch;

    if (bbUseDevelopmentBranch) {
      // Fetch Bitbucket development branch
      developmentBranch = (
        await bitbucketHttp.getJson<RepoBranchingModel>(
          `/2.0/repositories/${repository}/branching-model`
        )
      ).body.development?.branch?.name;

      if (developmentBranch) {
        config.defaultBranch = developmentBranch;
      }
    }

```

I don't see where you are checking if it exists though, as described in the new docs?

Reply:
It's part of the returned object from calling[ Repository Branching Model API](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-branching-model/#api-repositories-workspace-repo-slug-branching-model-get)

> Always has a development property. development.branch contains the actual repository branch object that is considered to be the development branch. development.branch will not be present if it does not exist.


The response is guaranteed to always return a `development.name`, such as
```
{
    ...
    "development": {
        "name": "main"
    }
    ...
}
```
or 
```
{
    ...
    "development": {
        "name": "develop"
    }
    ...
}
```
or whatever the repository admin has specified....

But it will only return a `development.branch.name` if that branch exists....
eg:
```
{
    ...
    "development": {
        "name": "develop",
         "use_mainbranch": false,
          "branch": {
              "name": "develop",
              "target": {
                  "hash": "e6aded4501c954c435174f399492fb1a854be188"
              },
              "type": "branch"
          }
    }
    ...
}
 ```

Reply:
Gotcha